### PR TITLE
improve table column sizing

### PR DESCRIPTION
**Pull request by devoncarew**
- improve table column sizing; allow table columns without explicit sizes to resize to the available remaining table width
- use more explicit types in `ColumnData`
- remove an unused method from `ColumnData`

I'll wait until after https://github.com/flutter/devtools/pull/1979 lands in the off chance that these PRs will have merge conflicts.

### CODE REVIEW

File: packages/devtools_app/lib/src/flutter/table.dart [hunk 1]
```diff
@@ -95,43 +94,57 @@ class FlatTableState<T> extends State<FlatTable<T>>
...
-      itemCount: widget.data.length,
-      columns: widget.columns,
-      columnWidths: columnWidths,
-      autoScrollContent: widget.autoScrollContent,
-      rowBuilder: _buildRow,
-      sortColumn: widget.sortColumn,
-      sortDirection: widget.sortDirection,
-      onSortChanged: _sortDataAndUpdate,
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/flutter/table.dart:124]: remove types

File: packages/devtools_app/lib/src/flutter/table.dart [hunk 2]
```diff
@@ -918,10 +934,14 @@ class _TableRowState<T> extends State<TableRow<T>>
                       : Icons.expand_more,
                   size: defaultIconSize,
                 ),
-              const SizedBox(height: defaultRowHeight, width: 4.0),
-              Text(
-                column.title,
-                overflow: TextOverflow.ellipsis,
+              if (isSortColumn) const SizedBox(width: 4.0),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/flutter/table.dart:937]: use densePadding const

File: packages/devtools_app/lib/src/flutter/table.dart (hunk 1 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/flutter/table.dart:124]: will do

File: packages/devtools_app/lib/src/flutter/table.dart (hunk 2 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/flutter/table.dart:937]: done

**Pull request merged**